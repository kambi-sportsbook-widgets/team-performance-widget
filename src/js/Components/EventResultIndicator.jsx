import React from 'react';
import PropTypes from 'prop-types';
import { translationModule } from 'kambi-widget-core-library';

const EventResultIndicator = ({ result }) => {
   const t = translationModule.getTranslation.bind(translationModule);

   return <div className={`kw-eventresultindicator KambiWidget-box--${result}`}>{t(result)}</div>;
};

EventResultIndicator.propTypes = {
   /**
    * Result of the event
    */
   result: PropTypes.string.isRequired
};

export default EventResultIndicator;
