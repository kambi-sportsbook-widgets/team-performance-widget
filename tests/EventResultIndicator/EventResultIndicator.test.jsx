/* eslint-env jest */
import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import { shallow } from 'enzyme';
import EventResultIndicator from '../../src/js/Components/EventResultIndicator';

let renderer;

jest.mock('kambi-widget-core-library', () => ({
   translationModule: {
      getTranslation: (string) => {
         if (string === 'D') {
            return 'Defeat';
         }
         return string;
      }
   },
}));

describe('EventResultIndicator DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });
   it('renders correctly', () => {
      expect(renderer.render(
         <EventResultIndicator result='D' />
      )).toMatchSnapshot();
   });
});
