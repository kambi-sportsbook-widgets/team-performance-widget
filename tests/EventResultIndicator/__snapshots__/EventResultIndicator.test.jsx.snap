// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventResultIndicator DOM rendering renders correctly 1`] = `
<div
  className="kw-eventresultindicator KambiWidget-DPrimaryColor"
>
  <span
    className="KambiWidget-DSecondaryColor"
  >
    Defeat
  </span>
</div>
`;
